cmd/kubeadm/app/phases/controlplane/manifests.go

```diff
@@ -64,7 +64,7 @@ func GetStaticPodSpecs(cfg *kubeadmapi.ClusterConfiguration, endpoint *kubeadmap
 			ReadinessProbe:  staticpodutil.ReadinessProbe(staticpodutil.GetAPIServerProbeAddress(endpoint), "/readyz", int(endpoint.BindPort), v1.URISchemeHTTPS),
 			StartupProbe:    staticpodutil.StartupProbe(staticpodutil.GetAPIServerProbeAddress(endpoint), "/livez", int(endpoint.BindPort), v1.URISchemeHTTPS, cfg.APIServer.TimeoutForControlPlane),
 			Resources:       staticpodutil.ComponentResources("250m"),
-			Env:             kubeadmutil.GetProxyEnvVars(),
+			Env:             append(cfg.APIServer.ExtraEnvs, kubeadmutil.GetProxyEnvVars()...),
```

Comment:
this is not merging the proxy and user vars?

they will be: FOO, BAR, FOO

i think we should make the user vars take priority / override and remove duplicates